docs/proposals/parameterized-config-management-plugins.md

```diff
@@ -0,0 +1,502 @@
+
+As an Argo CD user, I would like to be able to parameterize manifests built by a CMP.
+
+For example, if the Argo CD administrator has installed a CMP which applies a last-mile kustomize overlay to a Helm
+repo, I would like to be able to pass values to the Helm chart without having to manually discover those parameter names
+(in other words, they should show up in the Application UI just like with a native Helm Application). I also shouldn't 
+have to ask my Argo CD admin to modify the CMP to accommodate the values as environment variables.
+
+### Implementation Details/Notes/Constraints
+
+#### Prerequisites
+
+Since this proposal is designed to increase CMP adoption, we need to make sure there aren't any bugs that make CMPs
+less robust than native tools.
+
+Bugs to fix:
+1. [#8145](https://github.com/argoproj/argo-cd/issues/8145) - `argocd app sync/diff --local` doesn't account for sidecar CMPs
+
+#### Terms
+
+* **Parameter definition**: an instance of a data structure which describes an individual parameter that may be applied
+  to a specific Application. (See the [schema](#parameter-definition-schema) below.)
+* **Parameters announcement**: a list of parameter definitions. (See the [schema](#parameters-announcement-schema) below.)
+
+  "Parameters" is plural because each "announcement" will be a list of multiple parameter definitions.
+* **Parameterized CMP**: a CMP which supports rich parameters (i.e. more than environment variables). A CMP is
+  parameterized if either of these is true:
+  1. its configuration includes the sections consumed by the default CMP server to generate parameters announcements
+  2. it is a fully customized CMP server which implements an endpoint to generate parameters announcements
+
+#### CMP config schema
+
+This proposal adds a new `parameters` key to the ConfigManagementPlugin config spec.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ConfigManagementPlugin
+metadata:
+  name: cmp-plugin
+spec:
+  version: v1.0
+  generate:
+    command: ["example.sh"]
+  discover:
+    fileName: "./subdir/s*.yaml"
+  # NEW KEY
+  parameters:
+    # The declarative announcement follows the parameters announcement schema. This is where a parameter description
+    # should go if it applies to all apps for this CMP.
+    announcement:
+      - name: example
+    # The (optional) generated announcement is combined with the declarative announcement (if present).
+    command: ["example-params.sh"]
+```
+
+The currently-configured parameters (if there are any) will be communicated to both `generate.command` and 
+`parameters.command` via an `ARGOCD_PARAMETERS` environment variable. The parameters will be encoded according to the 
+[parameters serialization format](#parameters-manifest--parameters-serialization-format) defined below.
+
+Passing the parameters to the `parameters.command` will allow configuration of parameter discovery. For example:
+
+```yaml
+plugin:
```

Comment:
It seems that this is the `Application` resource. If so, I'm not sure it is a good idea to manipulate fields in the `spec` section as it could potentially generate conflicts with the desired state (git). Maybe it is better to move the `parameters` attribute into the `status` section. 

Something like:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  ...
spec: 
  ...
status:
  cmp:
    last-applied-parameters:
      main:
      - name: ignore-helm-charts
        value: '["chart-a", "chart-b"]'
```